src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway/nvmeof-gateway.component.ts

```diff
@@ -2,43 +2,98 @@ import { Component } from '@angular/core';
 
 import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
 import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
-import { NvmeofGateway } from '~/app/shared/models/nvmeof';
 
 import { NvmeofService } from '../../../shared/api/nvmeof.service';
+import { CephServiceSpec } from '~/app/shared/models/service.interface';
+import { CephServiceService } from '~/app/shared/api/ceph-service.service';
+
+type ComboBoxItem = {
+  content: string;
+  serviceName: string;
+  selected?: boolean;
+};
+
+type Gateway = {
+  id: string;
+  hostname: string;
+  status: string;
+};
 
 @Component({
   selector: 'cd-nvmeof-gateway',
   templateUrl: './nvmeof-gateway.component.html',
   styleUrls: ['./nvmeof-gateway.component.scss']
 })
 export class NvmeofGatewayComponent {
-  gateways: NvmeofGateway[] = [];
+  gateways: Gateway[] = [];
   gatewayColumns: any;
   selection = new CdTableSelection();
+  gwGroups: ComboBoxItem[] = [];
+  groupService: string = null;
 
-  constructor(private nvmeofService: NvmeofService, public actionLabels: ActionLabelsI18n) {}
+  constructor(
+    private nvmeofService: NvmeofService,
+    private cephServiceService: CephServiceService,
+    public actionLabels: ActionLabelsI18n
+  ) {}
 
   ngOnInit() {
+    this.getGatewayGroups();
     this.gatewayColumns = [
       {
-        name: $localize`Name`,
-        prop: 'name'
+        name: $localize`Gateway ID`,
+        prop: 'id'
       },
       {
-        name: $localize`Address`,
-        prop: 'addr'
+        name: $localize`Host name`,
+        prop: 'hostname'
       },
       {
-        name: $localize`Port`,
-        prop: 'port'
+        name: $localize`Status`,
+        prop: 'status'
```

Comment:
do we have a status and hostname to show in the output of `/api/nvmeof/gateway`?

Reply:
This is just daemon status, similar to what one would see in the service list page.
We dont have status info from `/api/nvmeof/gateway` API as of now. Its just some metadata, like addr, port and gateway ID.

Moving forward we need to align the UI view and API with the CLI version which gives useful information about gateways:

```
           "gw-id": "client.nvmeof.nvmeof.default.ceph-node-01.kdcguk",
            "anagrp-id": 2,
            "num-namespaces": 0,
            "performed-full-startup": 1,
            "Availability": "AVAILABLE",
            "ana states": " 1: STANDBY ,  2: ACTIVE "
```

Reply:
FYI the gateway output from `/api/nvmeof/gateway`

```
{
  "cli_version": "",
  "version": "1.2.17",
  "name": "client.nvmeof.nvmeof.default.ceph-node-02.hybprc",
  "group": "default",
  "addr": "192.168.100.102",
  "port": "5500",
  "load_balancing_group": 1,
  "spdk_version": "24.01"
}

```

Reply:
alright, thanks